### PR TITLE
Add Conjur Support for FindByName, FindByTag

### PR DESCRIPTION
**Pull request by szh**
## Problem Statement

The Conjur provider does not yet support the FindByName and FindByTag features. It also doesn't support extracting JSON properties from secrets.

## Related Issue

CyberArk internal issue ID: CNJR-4505, CNJR-4510

## Proposed Changes

- Added support for FindByName and FindByTag via `GetAllSecrets()`
- Added support for extracting fields from JSON secrets using the Property attribute, via `GetSecret()`
- Added unit tests for `GetSecretMap()`
- Updated the "Stability and Support" doc page to:
  - List the Conjur provider as stable
  - Indicate support for FindByName and FindByTag
  - Add myself as a maintainer

## Checklist

- [x] I have read the [contribution guidelines](https://external-secrets.io/latest/contributing/process/#submitting-a-pull-request)
- [x] All commits are signed with `git commit --signoff`
- [x] My changes have reasonable test coverage
- [x] All tests pass with `make test`
- [x] I ensured my PR is ready for review with `make reviewable`

### CODE REVIEW

File: docs/provider/conjur.md [hunk 1]
```diff
@@ -21,11 +21,18 @@ If you set up your Conjur server with a self-signed certificate, we recommend th
 {% include 'conjur-ca-bundle.yaml' %}
 ```
 
-### External secret store with apiKey authentication
+### External secret store
```
Comment (benskelker) [inline docs/provider/conjur.md:24]: @szh - not sure if we need this section, does it anything that isn't stated above?

File: docs/provider/conjur.md [hunk 2]
```diff
@@ -21,11 +21,18 @@ If you set up your Conjur server with a self-signed certificate, we recommend th
...
+
+The Conjur provider is configured as an external secret store in ESO. The external secret store is a Kubernetes custom resource that defines the connection to the Conjur server. There are two methods to authenticate to Conjur that are supported by the Conjur provider:
+
+* `apikey` - Uses a Conjur `hostid` and `apikey` to authenticate with Conjur.
+* `jwt` - Uses a JWT token to authenticate with Conjur.
+
+#### Option 1: External secret store with apiKey authentication
```
Comment (benskelker) [inline docs/provider/conjur.md:31]: I'd set this as a level-three heading (`###`), and move the heading levels below up one

File: docs/provider/conjur.md [hunk 3]
```diff
@@ -93,20 +117,34 @@ You can use an external JWT issuer or the Kubernetes API server to create the to
...
+# WARNING: running the delete command will delete the secret store configuration
+#
+# If there is a need to delete the external secretstore
+# kubectl delete secretstore -n external-secrets conjur
+```
+
+### Define an external secret
 
-#### Step 2: Define an external secret
+Now that you have set up the Conjur provider secret store, you can define an external secret that will fetch one or more secrets from Conjur.
```
Comment (benskelker) [inline docs/provider/conjur.md:137]: `After you have configured the Conjur provider secret store, you can fetch secrets from Conjur.`

File: docs/provider/conjur.md [hunk 4]
```diff
@@ -93,20 +117,34 @@ You can use an external JWT issuer or the Kubernetes API server to create the to
...
 
-Save the external secret file as: `conjur-external-secret.yaml`
+Here is an example of an external secret that fetches a single secret from Conjur:
```
Comment (benskelker) [inline docs/provider/conjur.md:139]: `Here is an example of how to fetch a single secret from Conjur:`

File: docs/provider/conjur.md [hunk 5]
```diff
@@ -93,20 +117,34 @@ You can use an external JWT issuer or the Kubernetes API server to create the to
...
 
-!!!Important
-    Unless you are using a [ClusterSecretStore](../api/clustersecretstore.md), credentials must reside in the same namespace as the SecretStore.
+Save the external secret file as `conjur-external-secret.yaml`.
 
-##### Find by Name and Find by Tag
+#### Find by Name and Find by Tag
 
 The Conjur provider also supports the Find by Name and Find by Tag ESO features. This means that
 you can use a regular expression or tags to dynamically fetch secrets from Conjur.
```
Comment (benskelker) [inline docs/provider/conjur.md:150]: `fetch multiple secrets from Conjur.`

File: docs/provider/conjur.md [hunk 6]
```diff
@@ -120,20 +158,7 @@ to only the secrets that it needs to access. Besides for being security best pra
 also decrease the amount of resources that the Conjur provider needs to fetch from Conjur and decrease
 load on both the Conjur server and ESO.
```
Comment (benskelker) [inline docs/provider/conjur.md:159]: `This reduces the load on both the Conjur server and ESO.`

File: docs/provider/conjur.md [hunk 7]
```diff
@@ -146,10 +171,10 @@ kubectl apply -n external-secrets -f conjur-external-secret.yaml
 # kubectl delete externalsecret -n external-secrets conjur
 ```
 
-#### Step 5: Get the K8s secret
+### Step 5: Get the K8s secret
 
-* Log in to your Conjur server and verify that your secret exists
-* Review the value of your Kubernetes secret to verify that it contains the same value as the Conjur server
+* Log in to your Conjur server and verify that your secret exists.
```
Comment (benskelker) [inline docs/provider/conjur.md:176]: when all bullets are phrases (not full sentences), no full stops (periods for you)

File: docs/provider/conjur.md (hunk 2 above)
Comment (szh) [reply docs/provider/conjur.md:31]: My idea here is that the outline should look like this:
image
This way External secret store is an L3 heading, with the two options (apikey and jwt) as L4 headings.

File: docs/provider/conjur.md [hunk 8]
```diff
@@ -119,7 +132,32 @@ kubectl apply -n external-secrets -f conjur-secret-store.yaml
...
+The Conjur provider also supports the Find by Name and Find by Tag ESO features. This means that
+you can use a regular expression or tags to dynamically fetch multiple secrets from Conjur.
+
+```yaml
+{% include 'conjur-external-secret-find.yaml' %}
+```
+
+If you use these search feature, we strongly recommend that you limit the permissions of the Conjur host
+to only the secrets that it needs to access. Besides for being security best practice, reduces the load on
+both the Conjur server and ESO.
```
Comment (benskelker) [inline docs/provider/conjur.md:158]: @szh, i missed some stuff here```suggestion
If you use these features, we strongly recommend that you limit the permissions of the Conjur host
to only the secrets that it needs to access. This is more secure and it reduces the load on
both the Conjur server and ESO.
```

File: docs/provider/conjur.md (hunk 2 above)
Comment (benskelker) [reply docs/provider/conjur.md:31]: @szh, yep, makes sense. but adding a comment to the intro paragraph above

File: docs/provider/conjur.md [hunk 9]
```diff
@@ -132,7 +170,7 @@ kubectl apply -n external-secrets -f conjur-external-secret.yaml
 # kubectl delete externalsecret -n external-secrets conjur
 ```
 
-#### Step 5: Get the K8s secret
+### Step 5: Get the K8s secret
```
Comment (benskelker) [inline docs/provider/conjur.md:173]:
```suggestion
### Get the K8s secret
```

File: docs/provider/conjur.md [hunk 10]
```diff
@@ -21,11 +21,18 @@ If you set up your Conjur server with a self-signed certificate, we recommend th
 {% include 'conjur-ca-bundle.yaml' %}
 ```
 
-### External secret store with apiKey authentication
+### External secret store
+
+The Conjur provider is configured as an external secret store in ESO. The external secret store is a Kubernetes custom resource that defines the connection to the Conjur server. There are two methods to authenticate to Conjur that are supported by the Conjur provider:
+
+* `apikey` - Uses a Conjur `hostid` and `apikey` to authenticate with Conjur.
+* `jwt` - Uses a JWT token to authenticate with Conjur.
```
Comment (benskelker) [inline docs/provider/conjur.md:29]: maybe add links to the option titles. and just for consistency:
```suggestion
* `apikey`: uses a Conjur `hostid` and `apikey` to authenticate with Conjur
* `jwt`: uses a JWT to authenticate with Conjur.
```

**Comment by szh** [comment]
Rebased on main. Ready for review whenever you are @Skarlso

**Comment by Skarlso** [comment]
Dang it, why did this fail? 😭 There are no logs.

https://github.com/external-secrets/external-secrets/actions/runs/8852999760/job/24318219100?pr=3364

**Comment by Skarlso** [comment]
ah, found it

```

 
--
Command failed: go env GOPATH go: downloading go1.22 (linux/amd64) go: download go1.22 for linux/amd64: toolchain not available
 

<br class="Apple-interchange-newline">
```

**Comment by Skarlso** [comment]
@szh Please update `go 1.22` to `go 1.22.0` as one of our dependencies requires a correct toolchain version now.

**Comment by szh** [comment]
@Skarlso done

**Review by Skarlso — APPROVED** [review]

**Pull request merged**